bleak/backends/corebluetooth/client.py

```diff
@@ -112,7 +113,15 @@ def disconnect_callback():
         manager = self._central_manager_delegate
         logger.debug("CentralManagerDelegate  at {}".format(manager))
         logger.debug("Connecting to BLE device @ {}".format(self.address))
-        await manager.connect(self._peripheral, disconnect_callback, timeout=timeout)
+        try:
+            await manager.connect(
+                self._peripheral, disconnect_callback, timeout=timeout
+            )
+        except asyncio.TimeoutError as error:
+            raise BleakDeviceNotFoundError(
```

Comment:
I seems like corebluetooth has no real mechanism for detecting "device not found" errors. the connect function just receives a Timeout error. Can we assume that the device was not found in case of this?

Reply:
I think we should probably just leave this as a timeout. 

If we really want a "device not found" here, I suppose we could check if the peripheral is still in the list returned by `CBCentalManager.retrievePeripheralsWithIdentifiers_` before attempting to connect.

Reply:
but I think it is fine to just leave it as-is for now.

Reply:
> If we really want a "device not found" here, I suppose we could check if the peripheral is still in the list returned by `CBCentalManager.retrievePeripheralsWithIdentifiers_` before attempting to connect.

This is already done a few lines above where the device is searched using the bleak scanner. The catch for the TimeoutError was only for the edge-case in which the device goes out of sight in between.
But I reverted this change as it might be misleading if the connection attempt times out for other reasons.